examples/run_CPMG.py

```diff
@@ -10,116 +10,131 @@
 # }}}
 # {{{create filename and save to config file
 date = datetime.now().strftime("%y%m%d")
 config_dict["type"] = "CPMG"
 config_dict["date"] = date
 config_dict["cpmg_counter"] += 1
-filename = f"{config_dict['date']}_{config_dict['chemical']}_{config_dict['type']}"
+filename = (
+    f"{config_dict['date']}_{config_dict['chemical']}_generic_{config_dict['type']}"
+)
+# }}}
+# {{{let computer set field
+print(
+    "I'm assuming that you've tuned your probe to",
+    config_dict["carrierFreq_MHz"],
+    "since that's what's in your .ini file",
+)
+Field = config_dict["carrierFreq_MHz"] / config_dict["gamma_eff_MHz_G"]
+print(
+    "Based on that, and the gamma_eff_MHz_G you have in your .ini file, I'm setting the field to %f"
+    % Field
+)
+with xepr() as x:
+    assert Field < 3700, "are you crazy??? field is too high!"
+    assert Field > 3300, "are you crazy?? field is too low!"
+    Field = x.set_field(Field)
+    print("field set to ", Field)
 # }}}
 # {{{set phase cycling
 phase_cycling = True
 if phase_cycling:
-    ph1_cyc = r_[0, 1, 2, 3]
-    nPhaseSteps = 4
+    ph_overall = r_[0, 1, 2, 3]
+    ph_diff = r_[0, 2]
+    ph1_cyc = array([(j + k) % 4 for k in ph_overall for j in ph_diff])
+    ph2_cyc = array([(k + 1) % 4 for k in ph_overall for j in ph_diff])
+    nPhaseSteps = len(ph_overall) * len(ph_diff)
 if not phase_cycling:
-    ph1_cyc = 0.0
     nPhaseSteps = 1
 # }}}
-# {{{better tau
-marker = 1.0  # 10/10/22 → what is this? → pretty sure the time needed to execute the marker command
-pad_start = config_dict["tau_extra_us"] - config_dict["deadtime_us"]
-pad_end = (
-    config_dict["tau_extra_us"] - config_dict["deblank_us"] - marker
-)  # marker + deblank
-assert(pad_start > 0), "tau_extra_us must be set to more than deadtime and more than deblank!"
-assert (pad_end > 0), "tau_extra_us must be set to more than deadtime and more than deblank!"
-twice_tau_echo_us = (  # the period between 180 pulses
-    config_dict["tau_extra_us"] * 2 + config_dict["acq_time_ms"] * 1e3
-)
-# now twice_tau_echo_us/2.0 is τ_echo, so I need to subtract the extra delays
-# imposed by the ppg to determine the remaining τ that I need
-config_dict["tau_us"] = twice_tau_echo_us / 2.0 - (
-    2
-    * config_dict["p90_us"]
-    / pi  # evolution during pulse -- see eq 6 of coherence paper
-    + config_dict["deadtime_us"]  # following 90
-    + config_dict["deblank_us"]  # before 180
-)
+prog_p90_us = prog_plen(config_dict["p90_us"])
+prog_p180_us = prog_plen(2 * config_dict["p90_us"])
+# {{{calculate symmetric tau by dividing 2tau by 2
+marker_us = 1.0
+config_dict["tau_us"] = (2*config_dict['deadtime_us']+1e3*config_dict['echo_acq_ms'])/2
+assert (config_dict['tau_us'] > prog_p180_us/pi + marker_us + config_dict['deblank_us'])
+assert (config_dict['deadtime_us'] > config_dict['deblank_us'] + 2*marker_us)
+assert (
+        2 * config_dict['deadtime_us']+1e3*config_dict['echo_acq_ms'] ==
+        2*config_dict['tau_us'])
```

Comment:
since you're explicitly setting this above, I think it's OK to delete this and bring down the line where you set this above

Reply:
deleted but will not bring line down since there is another assert statement above that involves the setting of tau (`assert (config_dict['tau_us']>2*prog_p90_us/pi+marker_us+config_dict['deblank_us'])`